metadata/src/main/java/org/apache/kafka/image/ImageWriter.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.image;
+
+import org.apache.kafka.common.protocol.ApiMessage;
+import org.apache.kafka.server.common.ApiMessageAndVersion;
+
+
+/**
+ * Writes out a metadata image.
+ */
+public interface ImageWriter extends AutoCloseable {
+    /**
+     * Write a record.
+     *
+     * @param version       The version of the record to write out.
+     *                      For convenience, this is an int.
+     * @param message       The message of the record to write out.
+     */
+    default void write(int version, ApiMessage message) {
```

Comment:
It's convenient because writing `(short) 0` and so on is annoying, and Java's typechecker can't recognize that 0 fits in a short apparently (or at least not the version of Java we are using currently)

> If we replace the literal zero with a symbol that has a type of short, do we still need this convenience?

we are probably not going to be using constants here (see above comment)